testing.go

```diff
@@ -8,7 +8,19 @@ import (
 	"testing"
 )
 
+type Test struct {
+	ctx  context.Context
+	span opentracing.Span
+	t    *testing.T
+}
+
 func InstrumentTest(t *testing.T, f func(ctx context.Context, t *testing.T)) {
+	test := StartTest(t)
+	defer test.End()
```

Comment:
I thought you were going to return both `test` and `ctx` as a separated variables. But I guess is cleaner this way.
Also the `InstrumentTest` function kept the `func` as parameter?